docs/admin/etcd.md

```diff
@@ -1,49 +1,163 @@
 ---
 assignees:
-- lavalamp
-title: Configuring Kubernetes Use of etcd
+- mml
+- wojtek-t
+title: Upgrading and Rolling Back etcd
 ---
 
+## About etcd
+
 [etcd](https://coreos.com/etcd/docs/latest/) is a highly-available key value
 store which Kubernetes uses for persistent storage of all of its REST API
-objects.
+objects.  For the mechanics behind how kubernetes builds, distributes and
+deploys Kubernetes, see _some doc_.
+
+## Background
+
+As of 1.5.1 kubernetes release, we are still using etcd from 2.2.1 release with
+v2 API.  Also, we have no pre-existing process for updating etcd, as we have
+never updated etcd by either minor or major version.
+
+Note that we need to migrate both: etcd versions that we are using (from 2.2.1
+to at least 3.0.x) as well as the version of API Kubernetes talks to etcd (3.0.x
+etcd binaries support both v2 and v3 API).
+
+This document describes how to do this migration.
+
+### Etcd upgrade limitations
+
+The way etcd was designed introduces some significant limitations on how the
+upgrade can be done. These are the main limitations. 
+
+#### One minor release at a time
+
+It is possible to upgrade only by one minor release at a time (though, within
+patch releases you can switch versions at any time back and forth). That
+basically means, that we cannot upgrade directly e.g. from 2.1.x to 2.3.x.
+
+Fortunately, this limitation it is easy to workaround - it is enough to start a
+cluster for any intermediate minor release, wait until it is healthy and
+functional and stop it then. This will do the migration itself underneath. 
+
+As an example: to upgrade from 2.1.x to 2.3.y version, it is enough to start
+etcd in 2.2.z version, wait until it is healthy, stop it and then start then
+2.3.y version, wait until it is up and we are done.
+
+#### No rollback
+
+Etcd doesn’t support rollback procedure at all (by design). That means, that if
+you migrate your cluster e.g. from 2.2.x to 2.3.y, there is no way to get back
+to 2.2.x (other than restoring from backup data from the moment when 2.2.x was
+running, though if we were running 2.3.y for some time, we will lose all data
+that were written in the meantime).
+
+This is a very huge limitation - we really need a rollback procedure if we face
+some problems with the new release.
+
+To make things better, we were provided a custom rollback tool by CoreOS (etcd
+folks), but:
+
+* This is a custom tool, which means it doesn’t have any guarantees and only
+  best-effort support from etcd team (though they fixed some bugs that we found
+  in the meantime).
+
+* The rollback can be done only from 3.0.x version (that is using v3 API) to
+  2.2.1 version (that is using v2 API).
```

Comment:
This is accurate, yes.  What @wojtek-t meant by that comment was that we don't need to change images.  The 3.0.17 image also contains the 2.2.1 binaries (and 2.3.7 and the rollback tool!).